File: CSharp/DependencyBreakingKatas/ExtractAndOverrideCall/Receipt.cs
using System;
using System.Collections.Generic;

namespace Org.Codecop.Dependencies.ExtractAndOverrideCall
{
    public class Receipt
    {
        public Money Amount { get; set; }
        public Money Tax { get; set; }
        public Money Total { get; set; }

        public IEnumerable<string> Format()
        {
            return new List<string>() { //
                    "Receipt", //
                    "=======", //
                    "Item 1 ... " + Amount.Format(), //
                    "Tax    ... " + Tax.Format(), //
                    "----------------", //
                    "Total  ... " + Total.Format() //
            };
        }
    }
}

File: TypeScript/app/c/receipt.ts
import { Money } from "./money";

export class Receipt {

    amount: Money;
    tax: Money;
    total: Money

    format(): string[] {
        return [ //
            "Receipt", //
            "=======", //
            "Item 1 ... " + this.amount.format(), //
            "Tax    ... " + this.tax.format(), //
            "----------------", //
            "Total  ... " + this.total.format() //
        ];
    }

}
